src/ResourceManager/DataFactoryV2/Commands.DataFactoryV2/IntegrationRuntimes/RemoveAzureDataFactoryIntegrationRuntimeCommand.cs

```diff
@@ -36,6 +45,26 @@ public override void ExecuteCmdlet()
             this.ByResourceId();
             this.ByIntegrationRuntimeObject();
 
+            if (Link.IsPresent && string.IsNullOrWhiteSpace(LinkedDataFactoryName))
```

Comment:
You can eliminate these checks by creating a new parameter set where both Link and LinkedDataFactoryName are mandatory, and LinkedDataFactoryName has a [ValidateNotNullOrEmpty] tag

Reply:
The cmdlet already has three Parameter sets:
ParameterSet: `ByResourceId`
    -`ResourceId` (unique/mandatory)
ParameterSet: `ByIntegrationRuntimeName`
    -`ResourceGroupName` (unique/mandatory)
    -`DataFactoryName` (unique/mandatory)
    -`Name` (unique/mandatory)
ParameterSet: `ByIntegrationRuntimeObject`
    -`InputObject` (unique/mandatory)

The new parameter set:
ParameterSet: `ByLinkedIntegrationRuntime`
    -`ResourceId` (optional)
    -`ResourceGroupName`(optional)
    -`DataFactoryName`(optional)
    -`Name`(optional)
    -`InputObject`(optional)
    -`Links `(unique/mandatory)
    -`LinkedDataFactoryName` (unique/mandatory)

To remove a linked data factory, the cmdlet is fine:
`Remove-AzureRmDataFactoryV2IntegrationRuntime -ResourceId $id -Links -LinkedDataFactoryName $linkedDataFactoryName`

However, to remove a normal IR, the parameter set of the cmdlet can not be determined:
`Remove-AzureRmDataFactoryV2IntegrationRuntime -ResourceId $id`

This is because '-ResourceId' is in two parameter set.
How do we resolve this issue with parameter set? @maddieclayton

Reply:
Actually, now that I look at it again, why do you need the -Link parameter at all? What would happen if you just added -LinkedDataFactoryName as an optional parameter to all three parameter sets, and followed the logic for removing Linked Data Factories if this parameter was provided?  I don't see any addition functionality provided by -Link

Reply:
@maddieclayton Agreed. Will remove -Links.